course-content/02_core_data_structure_concepts.qmd

```diff
@@ -230,6 +234,52 @@ What would these expressions return?
 7. If a part of the slice is out of range, the operation does not fail. `full_name[:]` gives the same result as `full_name[0:42]`, and the same result as `full_name[:103]`.
 :::
 
+### Dictionaries
+
+Another object that exists in Python is the `dictionary`. It is similar to a list in that it can hold a variety of different types of objects inside of it. However an important difference is in how we access these objects. With a list (or string), we have an ordered arrangement of items that we access with an **integer index**. However, we access the **values** in a dictionary with a **key**, which can be anything we want.
+
+Let's build a dictionary, which is denoted in Python with curly `{}` brackets:
+
+```{python}
+my_dict = {
+    'first_key': 'some value',
+    'A': ['a', 'differerent', 'type', 'of', 'object'],
+    2: False
+}
+
+print(my_dict)
+```
+
+Here we listed three **key - value** pairs. The key comes before the value, with a colon between. Commas separate different pairs. Now that we have a dictionary, we access it the same way as a list, with square `[]` brackets:
+
+```{python}
+print(my_dict['first_key'])
+print()
+print(my_dict['A'])
+print()
+print(my_dict[2])
+```
+
+Unlike a list, dictionaries are unordered, and so we cannot perform integer indexing or slicing of these elements:
+
+```{python}
+#| error: true
+my_dict[0]
+```
+
+```{python}
+#| error: true
+my_dict[0:5]
+```
+
+Dictionaries are an abstract data type that can take a while to get used to! They can be a powerful tool in Python. Common uses for dictionaries include:
+
+ * Creating searchable parameter lists for models
+ * Supplying extra arguments to functions 
+ * Storing complex outputs or datasets
```

Comment:
Are lookup tables also a common use?

Reply:
Yeah you could say that, it basically is just a fancy lookup table.